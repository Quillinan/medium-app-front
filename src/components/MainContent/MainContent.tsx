const MainContent: React.FC<{ currentTab: string }> = ({ currentTab }) => (
  <main>
    <div className='mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8'>
      <h2 className='text-2xl'>{currentTab} Content</h2>
      {/* Aqui você pode adicionar conteúdo específico para cada aba */}
    </div>
  </main>
);

export default MainContent;
