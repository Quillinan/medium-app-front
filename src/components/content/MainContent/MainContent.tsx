import BirthdayContent from '../BirthdayContent/BirthdayContent';

const MainContent: React.FC<{ currentTab: string }> = ({ currentTab }) => {
  const renderContent = () => {
    switch (currentTab) {
      case 'Criar Post':
        return <div>Conteúdo da aba Criar Post</div>;
      case 'Aniversários':
        return <BirthdayContent />;
      default:
        return <div>Conteúdo da aba Post</div>;
    }
  };

  return (
    <main data-testid='main-content'>
      <div className='mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8'>
        {renderContent()}
      </div>
    </main>
  );
};

export default MainContent;
