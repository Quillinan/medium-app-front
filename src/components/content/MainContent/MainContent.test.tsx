import { render, screen } from '@testing-library/react';
import MainContent from './MainContent';

describe('MainContent', () => {
  it('should render Menu content when no currentTab is provided', () => {
    render(<MainContent currentTab='' />);
    const defaultContent = screen.getByText(/Conteúdo da aba Menu/i);
    expect(defaultContent).toBeInTheDocument();
  });

  it('should render Menu content by default when no matching tab is provided', () => {
    render(<MainContent currentTab='UnknownTab' />);
    const defaultContent = screen.getByText(/Conteúdo da aba Menu/i);
    expect(defaultContent).toBeInTheDocument();
  });

  it('should render the correct content based on currentTab prop', () => {
    const { rerender } = render(<MainContent currentTab='Dashboard' />);
    expect(screen.getByText(/Conteúdo da aba Dashboard/i)).toBeInTheDocument();

    rerender(<MainContent currentTab='Aniversários' />);
    expect(screen.getByTestId('birthday-content')).toBeInTheDocument();

    rerender(<MainContent currentTab='UnknownTab' />);
    expect(screen.getByText(/Conteúdo da aba Menu/i)).toBeInTheDocument();
  });

  it('should render Dashboard content when currentTab is Dashboard', () => {
    render(<MainContent currentTab='Dashboard' />);
    const dashboardContent = screen.getByText(/Conteúdo da aba Dashboard/i);
    expect(dashboardContent).toBeInTheDocument();
  });

  it('should render BirthdayContent component when currentTab is Aniversários', () => {
    render(<MainContent currentTab='Aniversários' />);
    const birthdayContent = screen.getByTestId('birthday-content');
    expect(birthdayContent).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<MainContent currentTab='Dashboard' />);
    expect(asFragment()).toMatchSnapshot();
  });
});
