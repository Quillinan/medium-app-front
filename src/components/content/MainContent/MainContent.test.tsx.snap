// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainContent > should match snapshot for "UnknownTab" tab 1`] = `
<DocumentFragment>
  <main
    class="flex justify-center items-center h-full"
    data-testid="main-content"
  >
    <div
      class="flex items-center justify-center h-full mt-9"
      data-testid="posts-content"
    >
      <div
        class="space-y-8"
      >
        <p>
          Nenhum post encontrado.
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
